### PR TITLE
feat: add pipeline step for bitbucket

### LINKED ISSUE #4340: [Epic][bitbucket] some new feature transfer to bitbucket
## What and why to refactor
1. add transformationRule for bitbucket
2. add scope
3. support create bp with bitbucket in config-ui
4. support local bitbucket
5. support time fitler
6. support increment
7. https://github.com/apache/incubator-devlake/issues/4294

### PR DESCRIPTION
**Pull request by likyh**
### Summary
1. collect pipeline step as cicd task
2. add `page,pagelen,size` to fix a bug(to collect all page).

### Does this close any open issues?
Closes #4340

### Screenshots
Result like this:
```
bitbucket:BitbucketPipelineStep:1:{257bf6fd-4411-472b-82cf-0291d73d53ed},2023-02-20 12:09:33.981830 +00:00,2023-02-20 12:09:33.981830 +00:00,"{""ConnectionId"":1,""FullName"":""likyh/likyhphp""}",_raw_bitbucket_pipeline_steps,11670,"",staging,{990e12db-7802-4612-9713-b998dc16f468},SUCCESS,DONE,DEPLOYMENT,STAGING,8,2023-02-20 09:30:08.751166 +00:00,2023-02-20 09:30:16.433231 +00:00,""
```

### CODE REVIEW

File: backend/plugins/bitbucket/models/deployment.go [hunk 1]
```diff
@@ -23,21 +23,24 @@ import (
...
-	CompletedOn    *time.Time
-	LastUpdateTime *time.Time
+	ConnectionId    uint64 `gorm:"primaryKey"`
+	BitbucketId     string `gorm:"primaryKey"`
+	PipelineId      string `gorm:"type:varchar(255)"`
+	StepId          string `gorm:"type:varchar(255)"`
+	Type            string `gorm:"type:varchar(255)"`
+	Name            string `gorm:"type:varchar(255)"`
+	Environment     string `gorm:"type:varchar(255)"`
+	EnvironmentType string `gorm:"type:varchar(255)"`
```
Comment (likyh) [inline backend/plugins/bitbucket/models/deployment.go:33]: add StepId / Environment / EnvironmentType

File: backend/plugins/bitbucket/tasks/pipeline_steps_convertor.go [hunk 2]
```diff
@@ -27,74 +27,92 @@ import (
...
+			domainTask := &devops.CICDTask{
 				DomainEntity: domainlayer.DomainEntity{
-					Id: pipelineIdGen.Generate(data.Options.ConnectionId, bitbucketDeployment.BitbucketId),
+					Id: pipelineIdGen.Generate(data.Options.ConnectionId, bitbucketPipelineStep.BitbucketId),
 				},
-				Name: didgen.NewDomainIdGenerator(&models.BitbucketPipeline{}).
-					Generate(data.Options.ConnectionId, bitbucketDeployment.Name),
-				PipelineId: bitbucketDeployment.PipelineId,
+				Name:       bitbucketPipelineStep.Name,
+				PipelineId: bitbucketPipelineStep.PipelineId,
```
Comment (warren830) [inline backend/plugins/bitbucket/tasks/pipeline_steps_convertor.go:72]: we should use domain if of cicd pipeline

**Review by warren830 — APPROVED** [review]
Only a minor issue

**Review by warren830 — CHANGES_REQUESTED** [review]
A minor issue need to be fixed

File: backend/plugins/bitbucket/tasks/pipeline_steps_convertor.go (hunk 2 above)
Comment (likyh) [reply backend/plugins/bitbucket/tasks/pipeline_steps_convertor.go]: image
fixed

**Review by warren830 — APPROVED** [review]
LGTM

**Pull request merged**